### PR TITLE
1967 ratings

### PR DESCRIPTION
**Pull request by chrsin**
Pull request to use the new ranking properties from the server. Issue number is #1967

### CODE REVIEW

File: src/main/java/com/faforever/client/game/CreateGameController.java [hunk 1]
```diff
@@ -386,13 +387,30 @@ public void onCreateButtonClicked() {
         .map(ModVersion::getUid)
         .collect(Collectors.toSet());
 
+    int minRating = 0;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/game/CreateGameController.java]: There are also negative Ratings!

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/coop/CoopController.java [hunk 2]
```diff
@@ -267,7 +267,10 @@ public void onPlayButtonClicked() {
     modService.getFeaturedMod(COOP.getTechnicalName())
         .thenAccept(featuredModBean -> gameService.hostGame(new NewGameInfo(titleTextField.getText(),
             Strings.emptyToNull(passwordTextField.getText()), featuredModBean, getSelectedMission().getMapFolderName(),
-            emptySet())));
+            emptySet(),
+            0, //Rating isn't interesting for coop games
+            3000,
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/coop/CoopController.java]: constants like 0 and 3000 need to go away... should be something like null, null

File: src/main/java/com/faforever/client/game/CreateGameController.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/game/CreateGameController.java]: rating range is -invinity to +invinity

File: src/main/java/com/faforever/client/game/CreateGameController.java [hunk 3]
```diff
@@ -386,13 +387,30 @@ public void onCreateButtonClicked() {
...
+    int maxRating = 3000;
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/CreateGameController.java]: Nowhere should constants be found that imply a max or a min rating.... There is no max and no min rating. If no max rating is set then there is no max rating....

File: src/main/java/com/faforever/client/game/GameRangePredicate.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+public class GameRangePredicate implements Predicate<Game> {
+  private final Optional<Player> player;
+
+  public GameRangePredicate(Optional<Player> player) {
+    this.player = player;
+  }
+
+  @Override
+  public boolean test(Game game) {
+    if(!player.isPresent() || !game.getEnforceRating())
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/GameRangePredicate.java]: {} always for code style reasons

File: src/main/java/com/faforever/client/game/GameRangePredicate.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      return true;
+
+    float rating = RatingUtil.getGlobalRating(player.get());
+    return game.getMaxRating() >= rating && game.getMinRating() <= rating;
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/GameRangePredicate.java]: null checks!!!!

File: src/main/java/com/faforever/client/game/NewGameInfo.java [hunk 6]
```diff
@@ -17,8 +17,11 @@
   private String map;
   private Set<String> simMods;
   private GameVisibility gameVisibility;
+  private Integer ratingMin;
+  private Integer ratingMax;
+  private Boolean enforceRatingRange;
 
-  public NewGameInfo(String title, String password, FeaturedMod featuredMod, String map, Set<String> simMods) {
-    this(title, password, featuredMod, map, simMods, GameVisibility.PUBLIC);
+  public NewGameInfo(String title, String password, FeaturedMod featuredMod, String map, Set<String> simMods, int minRating, int maxRating, boolean enforceRating) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/NewGameInfo.java]: use Integer so you can set it to null

File: src/test/java/com/faforever/client/game/GameRangePredicateTest.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+@RunWith(MockitoJUnitRunner.class)
+public class GameRangePredicateTest {
+
+  private Player validPlayer;
+  private GameRangePredicate predicate;
+  private Game game;
+
+  @Before
+  public void setUp() {
+    validPlayer = new Player("");
```
Comment (1-alex98) [inline src/test/java/com/faforever/client/game/GameRangePredicateTest.java]: I think there is a PlayerBuilder

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/game/CreateGameController.java (hunk 1 above)
Comment (chrsin) [reply src/main/java/com/faforever/client/game/CreateGameController.java]: Good point! I'm adding a boolean parameter to the JavaFxUtil method, that signifies if negative values should be allowed in the text field.

File: src/main/java/com/faforever/client/coop/CoopController.java (hunk 2 above)
Comment (chrsin) [reply src/main/java/com/faforever/client/coop/CoopController.java]: Good point! I'll implement that.

File: src/main/java/com/faforever/client/game/GameRangePredicate.java (hunk 5 above)
Comment (chrsin) [reply src/main/java/com/faforever/client/game/GameRangePredicate.java]: Yeah now I'm changing from int to Integers I will need those. Or am I missing something else?

File: src/test/java/com/faforever/client/game/GameRangePredicateTest.java (hunk 7 above)
Comment (chrsin) [reply src/test/java/com/faforever/client/game/GameRangePredicateTest.java]: For these tests only the rating is relevant so don't think using the player builder gives any value. Do you disagree?

File: src/main/java/com/faforever/client/game/CustomGamesController.java [hunk 8]
```diff
@@ -179,9 +185,11 @@ private void updateFilteredItems() {
...
     boolean showModdedGames = showModdedGamesCheckBox.isSelected();
 
-    filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE.and(gameInfoBean ->
-        (showPasswordProtectedGames || !gameInfoBean.isPasswordProtected())
-            && (showModdedGames || gameInfoBean.getSimMods().isEmpty())));
+    filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE
+        .and(gameInfoBean ->
+            (showPasswordProtectedGames || !gameInfoBean.isPasswordProtected())
+                && (showModdedGames || gameInfoBean.getSimMods().isEmpty()))
+        .and(rangePredicate));
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/CustomGamesController.java]: No filtering needs to be done !!! The server already filters the game so they fit your rating. If the rating it not enforced the games are not filtered by the server and the rating only acts as a description.

File: src/main/java/com/faforever/client/game/GameRangePredicate.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
+package com.faforever.client.game;
+
+import com.faforever.client.player.Player;
+import com.faforever.client.util.RatingUtil;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class GameRangePredicate implements Predicate<Game> {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/GameRangePredicate.java]: this is already done by the server

File: src/main/resources/theme/play/create_game.fxml [hunk 10]
```diff
@@ -35,19 +35,19 @@
...
-        <Label alignment="CENTER_RIGHT" maxWidth="1.7976931348623157E308" text="%game.create.rating" textAlignment="JUSTIFY" GridPane.columnIndex="2" GridPane.rowIndex="1" />
-        <HBox alignment="CENTER" maxWidth="1.7976931348623157E308" spacing="10.0" GridPane.columnIndex="3" GridPane.columnSpan="2147483647" GridPane.rowIndex="1">
+        <HBox alignment="CENTER_RIGHT" spacing="10.0" maxWidth="1.7976931348623157E308" GridPane.columnIndex="2" GridPane.columnSpan="2147483647" GridPane.rowIndex="1">
             <children>
+                <Label alignment="CENTER_RIGHT" text="%game.create.rating" textAlignment="JUSTIFY" GridPane.columnIndex="2" GridPane.rowIndex="1" />
                 <TextField fx:id="minRankingTextField" minWidth="30.0" prefWidth="60.0" promptText="%games.create.minRating" HBox.hgrow="NEVER" />
                 <Label text="%games.create.minRating" />
-                <Pane minHeight="0.0" minWidth="0.0" HBox.hgrow="ALWAYS" />
+<!--                <Pane minHeight="0.0" minWidth="0.0" HBox.hgrow="ALWAYS" />-->
```
Comment (1-alex98) [inline src/main/resources/theme/play/create_game.fxml]: no commented out code

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/game/GameRangePredicate.java (hunk 9 above)
Comment (chrsin) [reply src/main/java/com/faforever/client/game/GameRangePredicate.java]: Makes sense... But also that makes me a sad panda.

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**